Realm/Realm.Weaver/Analytics/Analytics.cs

```diff
@@ -0,0 +1,846 @@
+            {
+                AnalyzeRealmClass(type);
+            }
+
+            AnalyzeTypeMethods(type);
+
+            foreach (var innerType in type.NestedTypes)
+            {
+                InternalAnalyzeSdkApi(innerType);
+            }
+        }
+
+        internal async Task<string> SubmitAnalytics()
+        {
+            try
+            {
+                var pretty = false;
+                var sendAddr = "https://data.mongodb-api.com/app/realmsdkmetrics-zmhtm/endpoint/metric_webhook/metric?data=";
+#if DEBUG
+                pretty = true;
+                sendAddr = "https://eu-central-1.aws.data.mongodb-api.com/app/realmmetricscollection-acxca/endpoint/realm_metrics/debug_route?data=";
+#endif
+                var payload = GetJsonPayload(pretty);
+
+                if (_config.AnalyticsCollection != AnalyticsCollection.DryRun)
+                {
+                    var base64Payload = Convert.ToBase64String(Encoding.UTF8.GetBytes(payload));
+                    await SendRequest(sendAddr, base64Payload, string.Empty);
+                }
+
+                return payload;
+            }
+            catch (Exception e)
+            {
+                _logger.Error($"Could not submit analytics.{Environment.NewLine}{e.Message}");
+                return e.Message;
+            }
+        }
+
+        private string GetJsonPayload(bool pretty)
+        {
+            var jsonPayload = new StringBuilder();
+
+            jsonPayload.Append('{');
+            if (pretty)
+            {
+                jsonPayload.AppendLine();
+            }
+
+            foreach (var entry in _realmEnvMetrics)
+            {
+                AppendKeyValue(entry);
+            }
+
+            foreach (var entry in _realmFeaturesToAnalyse)
+            {
+                AppendKeyValue(entry);
+            }
+
+            var trailingCommaIndex = pretty ? Environment.NewLine.Length + 1 : 1;
+            jsonPayload.Remove(jsonPayload.Length - trailingCommaIndex, 1);
+
+            jsonPayload.Append('}');
+
+            return jsonPayload.ToString();
+
+            void AppendKeyValue<Tkey, Tvalue>(KeyValuePair<Tkey, Tvalue> entry)
+            {
+                if (pretty)
+                {
+                    jsonPayload.Append('\t');
+                }
+
+                jsonPayload.Append($"\"{entry.Key}\": \"{entry.Value}\",");
+
+                if (pretty)
+                {
+                    jsonPayload.AppendLine();
+                }
+            }
+        }
+
+        private static async Task SendRequest(string prefixAddr, string payload, string suffixAddr)
+        {
+            using (var httpClient = new HttpClient())
+            {
+                httpClient.Timeout = TimeSpan.FromSeconds(4);
+                await httpClient.GetAsync(new Uri(prefixAddr + payload + suffixAddr));
+            }
+        }
+
+        private void AnalyzeTypeMethods(TypeDefinition type)
```

Comment:
A small thing, but I think it would make sense to have close related methods. So maybe first all the analysis methods, and then all the sending methods (including the json building)